tests/server/test_query_params.py

```diff
@@ -424,6 +424,45 @@ def test_list_has_and(self):
         expected_ids = ["mpf_1"]
         self._check_response(request, expected_ids, len(expected_ids))
 
+    def test_awkward_not_queries(self):
+        """ Test an awkward query from the spec examples. It should return all but 2 structures
+        in the test data. The test is done in three parts:
+
+            - first query the individual expressions that make up the OR,
+            - then do an empty query to get all IDs
+            - then negate the expressions and ensure that all IDs are returned except
+              those from the first queries.
+
+        """
+        expected_ids = ["mpf_3819"]
+        request = (
+            '/structures?filter=chemical_formula_descriptive="Ba2NaTi2MnRe2Si8HO26F"'
```

Comment:
See other (old) comment.
```suggestion
            '/structures?filter=chemical_formula_descriptive="Ba2NaTi2MnRe2Si8HO26F" AND chemical_formula_anonymous = "A26B8C2D2E2FGHI"'
```

Reply:
I'm with you now, fixed this